crates/runtime/src/actor/component/mod.rs

```diff
@@ -418,17 +418,37 @@ impl Instance {
 
     /// Instantiates and returns [`GuestBindings`] if exported by the [`Instance`].
     async fn as_guest_bindings(&mut self) -> anyhow::Result<GuestBindings> {
-        if let Ok((bindings, _)) =
-            guest_bindings::Guest::instantiate_async(&mut self.store, &self.component, &self.linker)
-                .await
+        let mut errors = Vec::new();
```

Comment:
How about we only store a singular `guest_error: Option<anyhow::Error>` here?

Reply:
Sure I can change to only storing the `guest_error` -- I think that's what most of the suggestions are about so will address them together

- [ ] store only guest error as an `Option` rather than the list